contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,231 @@
+You can integrate the Electron framework in your project using the Angular CLI and start building your application.
+This [article](https://medium.com/@yannmjl/how-to-build-native-cross-platform-desktop-apps-with-angular-electron-bd1d6e3919b2) is a good introduction to building desktop apps with Angular and Electron.
+
+### Benefit 3: Angular Provides Two-way Data Binding
+
+Web apps have two main components, a _View_ and a _Model_. The View component is responsible for displaying data (i.e., the part users see). The Model is concerned with the logic implementation of the application. It updates and retrieves the data to and from the database.
+_Data binding_ refers to the exchange of data between the View and Model components. There are two types of data binding as shown below.
+
+<center>
+<img src="data.jpg" width="50%">
+
+_Figure 2. Types of data binding_ <sup>[source](https://osmangoni.info/posts/one-way-and-two-way-data-binding/)</sup>
+</center>
+
+Some frameworks like React provide one-way data binding i.e., when there is a change to the Model, the View updates automatically.
+However, if the app allows users to change data through the view (e.g., entering your name in the textbox), the developer has to write code to propagate that change to the Model.
+
+In contrast, Angular provides data binding in both ways. That means even changes to the View can be propagated to the Model automatically, without writing extra code for that.
+
+This [article](https://medium.com/@preethi.s/angular-custom-two-way-data-binding-3e618309d6c7) on Medium provides a good introduction to two-way data binding in Angular.
+
+### Disadvantages of Angular
+
+Like any other framework/library, Angular has its share of disadvantages.
+
+1. **Steep Learning Curve** - Angular requires you to learn many concepts, such as directives, modules, components, services and many more. 
+The large number of new concepts can be confusing to newcomers. Additionally, Angular requires you to learn a new language, TypeScript.
+
+2. **Opinionated Framework** - Angular is opinionated about how you structure your code. This means that Angular provides you defaults for building applications. For instance, there are in-built services to perform data fetching, state management and much more. It is also opinionated about the development language as it requires you to use TypeScript.<br/>
+In contrast, React allows you to integrate any third party services in your application.
+    
+### Comparison with other popular frontend frameworks
+
+[React]("https://reactjs.org/") and [Vue]("https://vuejs.org/") are two popular alternatives to Angular.
+
+React is a popular Javascript library, open sourced by Facebook. The ease of learning React is a key advantage.
+It also provides more flexibility as it allows integration of third party libraries.
+Similarly, Vue offers some advantages, like flexibility, simple structure and ease of integration.
```

Comment:
This paragraph is half-baked. For example, you have this sentence about `React is a popular Javascript library, open sourced by Facebook.` but there is no equivalent for Vue. How do the three compare in terms of learning curve, flexibility etc. If you cannot do a three way comparison, at least do compare each to Angular so that both alternatives get equal treatment.